app/src/main/java/xyz/whereuat/whereuat/ui/adapters/PendingRequestAdapter.java

```diff
@@ -40,13 +56,91 @@ public void bindView (View view, Context context, Cursor cursor) {
         (view.findViewById(R.id.add_btn)).setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
-                Toast.makeText(mContext, "Adding contact...", Toast.LENGTH_SHORT).show();
+                if (name.isEmpty()) {
+                    handleWithAddContactDialog(number);
+                } else {
+                    addContact(context, name, number);
+                }
             }
         });
         (view.findViewById(R.id.delete_btn)).setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
-                Toast.makeText(mContext, "Deleting request...", Toast.LENGTH_SHORT).show();
+                handleDelete(context, number);
+            }
+        });
+    }
+
+    /**
+     * If the unknown contact has no name (meaning their number was not found in the phonebook),
+     * a dialog for entering their name should be shown before the contact is added to the database.
+     * This function displays the dialog and passes the contact's number to it. The dialog will
+     * handle actually making the database insertion.
+     *
+     * @param number the new contact's phone number
+     */
+    private void handleWithAddContactDialog(String number) {
+        FragmentManager fm = mActivity.getFragmentManager();
+        AddContactDialogFragment add_contact_dialog = new AddContactDialogFragment();
+        Bundle args = new Bundle();
+        args.putString(Constants.CONTACT_NUMBER_BUNDLE_EXTRA, number);
+        add_contact_dialog.setArguments(args);
+        add_contact_dialog.show(fm, TAG);
+    }
+
+    /**
+     * This function handles adding the contact to the database.
+     *
+     * @param context the context to execute the database insert from
+     * @param name the new contact's name
+     * @param number the new contact's number
+     */
+    private void addContact(final Context context, final String name, final String number) {
```

Comment:
Would it be better to have `AddContactDialogFragment` be a subclass of `PendingRequestAdapter` to prevent this double-declaration stuff? Could use a similar structure for the `KeyLocDialogFragment`. In any case, out of scope of this commit